modules/infotext/currency.lua

```diff
@@ -0,0 +1,101 @@
+-- Currency Infotext
+
+-- ####################################################################################################################
+-- ##### Setup and Locals #############################################################################################
+-- ####################################################################################################################
+
+local _, LUI = ...
+local module = LUI:GetModule("Infotext")
+local element = module:NewElement("Currency", "AceEvent-3.0", "AceHook-3.0")
+local L = LUI.L
+
+-- local copies
+local wipe, format, tconcat = wipe, format, table.concat
+local C_CurrencyInfo = C_CurrencyInfo
+
+-- Constants
+local MAX_WATCHED_TOKENS = MAX_WATCHED_TOKENS
+local CURRENCY_FORMAT = "%d\124T%s:%d:%d:2:0\124t"
+local CURRENCY = CURRENCY
+
+-- locals
+local currencyString = {}
+local getCurrencyInfo = C_CurrencyInfo.GetBackpackCurrencyInfo
+	if not LUI.IsRetail then
+		getCurrencyInfo = GetBackpackCurrencyInfo
+	end
+local getCurrencyInfoListSize = C_CurrencyInfo.GetCurrencyListSize
+	if not LUI.IsRetail then
+		getCurrencyInfoListSize = GetCurrencyListSize
+	end
```

Comment:
Part of me like the indent, but part of me is really thrown off by it.

It might be possible to to in-line those:
`func = (LUI.IsRetail) and C_CurrencyInfo.Func or Func`